src/murmur/Meta.cpp

```diff
@@ -747,8 +749,25 @@ void Meta::killAll() {
 	qhServers.clear();
 }
 
+/**
+ * If configured to not autoban successful connection attempts, we can remove those from being
+ * tracked with this method.
+ */
+void Meta::successfulConnectionFrom(const QHostAddress &addr) {
+	if (!mp.iBanSuccessful) {
+		QList<Timer> &ql = qhAttempts[addr];
+		// Seems like this is the most efficient way to clear the list, given:
+		// 1. ql.clear() allocates a new array
+		// 2. ql has less than iBanAttempts members
+		// 3. seems like ql.removeFirst() might actually copy elements to shift to the front
+		while (!ql.empty()) {
+			ql.removeLast();
+		}
+	}
+}
+
 bool Meta::banCheck(const QHostAddress &addr) {
-	if ((mp.iBanTries == 0) || (mp.iBanTimeframe == 0))
+	if ((mp.iBanTries <= 0) || (mp.iBanTimeframe == 0))
```

Comment:
Why is `<=` needed here?

Reply:
It's not, I can take it out if you want. It just seemed like negative values were non-sensical so it would be better to treat anything <= 0 as turning the feature off.

Reply:
Nah it's fine. I just didn't see a connection to your original intention - thus my question :)

As a matter of fact I think you can do the same to the check regarding the timeframe :point_up:

Reply:
Yeah, I felt guilty sneaking it in here with this PR, but I wasn't sure it was a big enough change to warrant its own PR. If you're happy with it here that works for me, or I can take it out. For now I went ahead and did the same to iBanTimeFrame.

Reply:
There's no such thing as too small for a PR - after all they're free :P

But I guess it's a small enough change to be sneaked in here. Ideally this would be a separate commit though, so if you have the time to split this that'd be great. If not I don't think it's a big deal either :)